sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/TestJsonData.scala

```diff
@@ -188,6 +188,12 @@ private[json] trait TestJsonData {
       """{"b":"str_b_4", "a":"str_a_4", "c":"str_c_4"}""" ::
       """]""" :: Nil)
 
+  def additionalCorruptRecords: RDD[String] =
+    sqlContext.sparkContext.parallelize(
+      """{"dummy":"test"}""" ::
+      """42""" ::
+      """     ","ian":"test"}""" :: Nil)
```

Comment:
Can you add a record like `"""[1,, 2, 3]"""` (a top level JSON array having and its elements are not JSON objects)?